File: lab-redes/java/udp/ClienteUDP.java
import java.net.*;
import java.util.Scanner;

public class ClienteUDP {
    public static void main(String[] args) throws Exception {
        String host = "localhost";
        int porta = 5001;

        try (DatagramSocket socket = new DatagramSocket()) {
            InetAddress endereco = InetAddress.getByName(host);
            Scanner teclado = new Scanner(System.in);
            byte[] buffer = new byte[1024];

            System.out.println("[UDP] Pronto para enviar. Digite 'sair' para encerrar.");
            while (true) {
                System.out.print("> ");
                String mensagem = teclado.nextLine();
                byte[] dados = mensagem.getBytes();

                DatagramPacket pacote = new DatagramPacket(dados, dados.length, endereco, porta);
                socket.send(pacote);
                if (mensagem.equalsIgnoreCase("sair")) break;

                DatagramPacket resposta = new DatagramPacket(buffer, buffer.length);
                socket.receive(resposta);
                System.out.println(new String(resposta.getData(), 0, resposta.getLength()));
            }

            teclado.close();
        }
    }
}

File: lab-redes/python/udp/cliente_udp.py
import socket

HOST = "localhost"
PORTA = 5001

with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as cliente:
    print("[UDP] Pronto para enviar. Digite 'sair' para encerrar.")
    while True:
        mensagem = input("> ")
        cliente.sendto(mensagem.encode("utf-8"), (HOST, PORTA))
        if mensagem.lower() == "sair":
            break
        dados, _ = cliente.recvfrom(1024)
        print(dados.decode("utf-8"))